File: clojure/day01/index1.clj
(require '[clojure.string :as s])

(def parseInt #(Integer/parseInt %))

(defn split [p s] (s/split s p))

(def input (->> "./input"
  slurp
  s/trim
  (split #"")
  (map parseInt)
  (into [])
))

(def l (count input))
(def input' (conj input (first input)))

(def res (loop [sum 0 i 0]
           (if (== i l)
              sum
              (recur 
                (+ sum (if-not (== (get input' i) (get input' (inc i))) 0 (get input' i)))
                (inc i)))))

(println res)

File: nodejs/day01/index1.js
const fs = require("fs");

const input = fs
  .readFileSync("./input", "utf-8")
  .trim()
  .split("")
  .map(d => parseInt(d, 10));

const l = input.length;
input.push(input[0]);

let sum = 0;
for (let i = 0; i < l; i++) {
  if (input[i] === input[i + 1]) {
    sum += input[i];
  }
}

console.log(sum);
